ament_cmake_test/cmake/run_test.py

```diff
@@ -78,43 +78,59 @@ def main(argv=sys.argv[1:]):
     with open(args.result_file, 'w') as h:
         h.write(failure_result_file)
 
-    print("-- run_test.py: invoking following command in '%s':\n - %s" %
-          (os.getcwd(), ' '.join(args.command)))
-
     # collect output / exception to generate more detailed result file
     # if the command fails to generate it
     output = ''
-    h = None
+    output_handle = None
     if args.output_file:
         output_path = os.path.dirname(args.output_file)
         if not os.path.exists(output_path):
             os.makedirs(output_path)
-        h = open(args.output_file, 'wb')
+        output_handle = open(args.output_file, 'wb')
+
+    def log(msg, **kwargs):
+        print(msg, **kwargs)
+        if output_handle:
+            output_handle.write((msg + '\n').encode())
+            output_handle.flush()
+
+    log("-- run_test.py: invoking following command in '%s':\n - %s" %
+        (os.getcwd(), ' '.join(args.command)))
+    if output_handle:
```

Comment:
Should you not just use log here? If there's separation in the log file, separation in the console output would be good too.